### DISCUSSION TITLE
Empty selection by clicking in it

### DISCUSSION THREAD

**Discussion by janpeterka** [josefarias/hotwire_combobox#149] [Q&A]
Hi,
I recently started using combobox, and there's one use-case I'm not sure about how to use.
There are cases when user selects some option, goes out, and then returns to change it. Currently they have to click the X, but I'd like to empty it on click inside the input.

I tried using `data-action="click->hw-combobox#clearOrToggleOnHandleClick"`, but that collapses the input.

Can there be `clear` method added, which just empties the input and option, that I'd be able to call?

Or is there some simpler way I'm missing?

Thanks a lot!

**Comment by ericlevy** [comment]
I'm running into this same issue, and I'm wondering if I'm overlooking a way to do this.  I have a dynamic nested form.  Basically you select an option from the first combobox (which is a list of Performers); and then a second combobox appears with a list of Events for the selected Performer.  If you then select an Event in the second combobox, and then change the Performer in the first combobox, I cannot figure out how to reset/clear what's shown in the combobox announcer in the second combobox (list of Events).  I'm assuming there's an easy-ish way to do this that I'm just not thinking of (as I'm a little new to turbo), so any suggestions would be much appreciated.

**Reply by janpeterka** [reply]
How do you implement showing the second combobox after first one is set? Stimulus? Turbo (submit on filling first combobox)?

**Comment by josefarias** [comment]
Hey @janpeterka @ericlevy sorry about the radio silence here. Much like https://github.com/josefarias/hotwire_combobox/discussions/174 I've been sitting with this one. Lot's of ways to go about it.

I don't think we have a way to do this out of the box just yet — we'll have to implement something at the library level. I've been strapped for time lately but I'm hoping to take a look soon.

**Comment by josefarias** [comment]
@janpeterka @ericlevy the next release will allow you to call `.clear()` on the combobox stimulus controller via outlets. The PR might serve as reference: https://github.com/josefarias/hotwire_combobox/pull/193

I think that should cover your use cases. But please let me know if not. Closing for now. Feel free to point your gem to the latest on `main` and try it out!

**Reply by janpeterka** [reply]
Amazing, thanks! Will try when I'll get to it (probably early September)
